dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -1301,6 +1300,42 @@ public JobExecutionEntity call() throws Exception {
         }
     }
 
+    @Override
+    public List<Long> getJobExecutionsRunning(final String jobName) {
+
+        final List<Long> runningInstances = getJobInstancesRunning(jobName);
+
+        final EntityManager em = getPsu().createEntityManager();
+        try {
+            List<Long> exec = new TranRequest<List<Long>>(em) {
+
+                @Override
+                public List<Long> call() throws Exception {
+
+                    List<Long> result = new ArrayList<Long>();
+
+                    for (int i = 0; i < runningInstances.size(); i++) {
+                        long instanceId = runningInstances.get(i);
+                        TypedQuery<Long> query = em.createNamedQuery(JobExecutionEntity.GET_JOB_EXECUTIONIDS_BY_JOB_INST_ID,
+                                                                     Long.class);
+                        query.setParameter("instanceId", instanceId);
+                        result.addAll(query.getResultList());
+                    }
```

Comment:
Should we try to do this in one query... search for ids "IN" the set instead of 1 query per loop iteration?   
Also previously we were sorting by createTime `ORDER BY e.createTime DESC`... are we still getting that result or should we do a later sort to ensure this?

Reply:
Revised this query per your suggestion in the latest commit